### PR TITLE
prov/gnix: more work on lower level structs

### PR DESCRIPTION
**Pull request by hppritcha**
Signed-off-by: Howard Pritchard howardp@lanl.gov

### CODE REVIEW

**Comment by hppritcha** [comment]
at the usual suspects

@bturrubiates 
@sungeunchoi

**Comment by hppritcha** [comment]
had to rebase to fix merge issues.

File: prov/gni/src/gnix.h [hunk 1]
```diff
@@ -218,6 +220,79 @@ struct gnix_fid_domain {
...
+	struct gnix_fid_domain *domain;
+	struct gnix_fid_cq *send_cq;
+	struct gnix_fid_cq *recv_cq;
+	struct gnix_fid_av *av;
+	struct gnix_nic *nic;
+	union {
+		void *vc_hash_hndl;  /*used for FI_AV_MAP */
+		void *vc_table;      /* used for FI_AV_TABLE */
+		void *vc;
+	} u;
```
Comment (bturrubiates) [inline prov/gni/src/gnix.h]: U for unnamed?

**Comment by hppritcha** [comment]
changes per feedback from @bturrubiates

**Pull request merged**